coalib/results/TextPosition.py

```diff
@@ -31,3 +46,35 @@ def line(self):
     @property
     def column(self):
         return self._column
+
+    @classmethod
+    def from_position(cls, text, position, create_object=True):
+        """
+        Makes a TextPosition object or a tuple containing (line, column)
+        by calculating line number and column in the text, from position.
+        Uses \\n as the newline character. Lines and columns start from 1.
+
+        :param text:        A string with all the contents of file.
+        :param pos_to_find: Position of character to be found in the
+                            (line, column) form.
+        :create_object:     Boolean which determines whether to return
+                            a TextPosition Object or a tuple.
+        :return:            a TextPosition object or a tuple of the form
+                            (line, column).
+        """
+        line = 1
+        try:
+            while True:
+                textline = text[line - 1]
+                new_position = position - len(textline)
+                if new_position < 0:
+                    break
+                line += 1
+                position = new_position
+            col = position+1
+            if create_object:
+                return cls(line, col, position=position)
```

Comment:
i guess so though i  didn't pass it since it'll call the function again, and do the same process twice(during assert) i wanted to get some insight on this